### PR TITLE
string literal enum support

### PR DESCRIPTION
**Pull request by maxcan**
Hacked up support for string literal enums.

This partially addresses #6

### CODE REVIEW

File: src/metadataGeneration/resolveType.ts [hunk 1]
```diff
@@ -62,6 +62,13 @@ function generateReferenceType(typeName: string, cacheReferenceType = true): Ref
     name: typeName,
     properties: properties
   };
+  if (modelTypeDeclaration.kind === ts.SyntaxKind.TypeAliasDeclaration) {
+    let innerType = modelTypeDeclaration.type;
```
Comment (lukeautry) [inline src/metadataGeneration/resolveType.ts]: Can this be a const?

File: src/metadataGeneration/resolveType.ts [hunk 2]
```diff
@@ -62,6 +62,13 @@ function generateReferenceType(typeName: string, cacheReferenceType = true): Ref
...
+    if (innerType.kind === ts.SyntaxKind.UnionType && (innerType as any).types) {
+      let unionTypes = (innerType as any).types;
```
Comment (lukeautry) [inline src/metadataGeneration/resolveType.ts]: const here too

**Comment by lukeautry** [comment]
@maxcan  Other than the couple of comments I made, this looks good to me. Thanks for digging in.

As far as I know, we could support literal types for numbers, but it would never make sense to supported mixed types based on the swagger specification.

File: src/metadataGeneration/resolveType.ts (hunk 2 above)
Comment (maxcan) [inline src/metadataGeneration/resolveType.ts]: done

File: src/metadataGeneration/resolveType.ts (hunk 1 above)
Comment (maxcan) [inline src/metadataGeneration/resolveType.ts]: done

**Comment by maxcan** [comment]
does it need a version bump?

**Comment by lukeautry** [comment]
I'll take care of the version bump.

**Pull request merged**